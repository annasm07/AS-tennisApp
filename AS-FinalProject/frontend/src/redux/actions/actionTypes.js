export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
