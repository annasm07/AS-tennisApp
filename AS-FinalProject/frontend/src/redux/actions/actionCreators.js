import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
