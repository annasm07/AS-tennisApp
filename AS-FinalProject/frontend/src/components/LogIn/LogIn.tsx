import React, {useEffect, useState} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import {connect} from 'react-redux';
import globalStyles from '../../theme/globalThemes';
import {logIn} from '../../redux/actions/actionCreators';

const LogIn = ({tokens, dispatch, navigation, user}: any) => {
  useEffect(() => {
    tokens.length === 2 && navigation.navigate('FixedNavigator');
  }, [tokens, navigation]);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  function handleLogIn() {
    dispatch(logIn(email, password));
  }
  return (
    <SafeAreaView style={styles.body}>
      <Text style={styles.title}>Log In</Text>
      <TextInput
        style={styles.input}
        onChangeText={userEmail => setEmail(userEmail)}
        placeholder="Email"
        defaultValue={email}
        autoCapitalize="none"
      />
      <TextInput
        style={styles.input}
        onChangeText={pass => setPassword(pass)}
        defaultValue={password}
        secureTextEntry={true}
        placeholder="Password"
        autoCapitalize="none"
      />
      <TouchableOpacity style={globalStyles.buttonYellow} onPress={handleLogIn}>
        <Text testID="LogInButton">Log in</Text>
      </TouchableOpacity>
      {user.name && <Text style={styles.signedUp}>Signed Up Successfully</Text>}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    top: '32%',
  },
  title: {
    textAlign: 'center',
    fontSize: 24,
    padding: 30,
  },
  input: {
    height: 40,
    margin: 8,
    borderWidth: 1,
    paddingLeft: 20,
    borderRadius: 20,
    fontSize: 16,
    marginHorizontal: 40,
    borderColor: '#7A7A7A',
  },
  signedUp: {
    color: '#70BA5D',
    alignSelf: 'center',
    marginTop: 60,
  },
});

function mapStateToProps({tokens, user}: any) {
  return {
    tokens,
    user,
  };
}

export default connect(mapStateToProps)(LogIn);
