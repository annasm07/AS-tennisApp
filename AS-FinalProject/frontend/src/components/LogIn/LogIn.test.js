import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import LogIn from './LogIn';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a LogIn component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}},
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'logIn').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });

  test('When typing on email input, then setEmail is invoked', () => {
    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <LogIn navigation={navigation} />
      </Provider>,
    );
    const emailInput = getByPlaceholderText('Email');
    let setEmail = jest.fn();
    fireEvent(emailInput, 'onChangeText', setEmail);
    expect(setEmail).toHaveBeenCalled();
  });

  test('When typing on password input, then setPassword is invoked', () => {
    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <LogIn navigation={navigation} />
      </Provider>,
    );
    const passwordInput = getByPlaceholderText('Password');
    let setPassword = jest.fn();
    fireEvent(passwordInput, 'onChangeText', setPassword);
    expect(setPassword).toHaveBeenCalled();
  });

  test('When Match is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <LogIn navigation={navigation} />
      </Provider>,
    );
    const LogInButton = getByTestId('LogInButton');
    fireEvent.press(LogInButton);
    expect(actions.logIn).toHaveBeenCalled();
  });
});

describe('When invoked a LogIn component and use.name exists', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    tokens: ['abc', 'cde'],
    user: {user: {playerId: '123cde'}, name: 'Anna'},
  });
  let navigation;
  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
    };
  });
  test('Should render sign in component', () => {
    const logIn = render(
      <Provider store={store}>
        <LogIn navigation={navigation} />
      </Provider>,
    );
    expect(logIn).toMatchSnapshot();
  });
});
