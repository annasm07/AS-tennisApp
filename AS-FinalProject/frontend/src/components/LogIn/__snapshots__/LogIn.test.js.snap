// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a LogIn component and use.name exists Should render sign in component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "top": "32%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "padding": 30,
        "textAlign": "center",
      }
    }
  >
    Log In
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    defaultValue=""
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#7A7A7A",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 40,
        "margin": 8,
        "marginHorizontal": 40,
        "paddingLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    defaultValue=""
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderColor": "#7A7A7A",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 40,
        "margin": 8,
        "marginHorizontal": 40,
        "paddingLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F5BF00",
        "borderColor": "#000000",
        "borderRadius": 20,
        "borderWidth": 1,
        "fontSize": 50,
        "marginTop": 30,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
        "width": 120,
      }
    }
  >
    <Text
      testID="LogInButton"
    >
      Log in
    </Text>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#70BA5D",
        "marginTop": 60,
      }
    }
  >
    Signed Up Successfully
  </Text>
</RCTSafeAreaView>
`;
